lib/ansible/modules/net_tools/netplan.py

```diff
@@ -0,0 +1,879 @@
+                        module.fail_json(msg='mode must be balance-tlb or balance-alb to define {0} param'.format(key))
+    if module.params['type'] == 'ethernets':
+        for key in module.params:
+            if module.params.get(key) is not None:
+                if key in BONDS:
+                    module.fail_json(msg='BONDS options can not be defined with ethernets Type')
+                if key in BRIDGES:
+                    module.fail_json(msg='BRIDGES options can not be defined with ethernets Type')
+                if key in VLANS:
+                    module.fail_json(msg='VLANS options can not be defined with ethernets Type')
+                if key in WIFIS:
+                    module.fail_json(msg='WIFIs options can not be defined with ethernets Type')
+    if module.params['type'] == 'vlans':
+        for key in module.params:
+            if module.params.get(key) is not None:
+                if key in BONDS:
+                    module.fail_json(msg='BONDS options can not be defined with vlans Type')
+                if key in BRIDGES:
+                    module.fail_json(msg='BRIDGES options can not be defined with vlans Type')
+                if key in WIFIS:
+                    module.fail_json(msg='WIFIs options can not be defined with vlans Type')
+    if module.params['type'] == 'wifis':
+        for key in module.params:
+            if module.params.get(key) is not None:
+                if key in BONDS:
+                    module.fail_json(msg='BONDS options can not be defined with vlans Type')
+                if key in BRIDGES:
+                    module.fail_json(msg='BRIDGES options can not be defined with vlans Type')
+                if key in VLANS:
+                    module.fail_json(msg='VLANS options can not be defined with ethernets Type')
+
+
+def get_netplan_dict(params):
+    netplan_dict = {'network': dict()}
+    # Define default netplan version and renderes both to 2 and networkd
+    if params.get('version'):
+        netplan_dict['network']['version'] = params.get('version')
+    else:
+        netplan_dict['network']['version'] = 2
+    if params.get('renderer'):
+        netplan_dict['network']['renderer'] = params.get('renderer')
+    else:
+        netplan_dict['network']['renderer'] = 'networkd'
+    netplan_dict['network'][params.get('type')] = dict()
+    netplan_dict['network'][params.get('type')][params.get('interface-id')] = dict()
+    for key in params:
+        if key not in SPECIFC_ANSIBLE_NETPLAN and params.get(key) is not None:
+            if key in MATCH:
+                match_option = '{0}'.format(key.split('match-')[1])
+                netplan_dict['network'][params.get('type')][params.get('interface-id')]['match'] = dict()
+                netplan_dict['network'][params.get('type')][params.get('interface-id')]['match'][match_option] = params.get(key)
+            # dhcp4-overrides
+            if key in DHCP_OVERRIDES:
```

Comment:
hey @lets00,
instead of "if", this should be "elif" and all the other "if"s below should be changed too.
Otherwise the last "else" condidition code block will be always executed, except for "WIFIS" params. 

Example output file for the parameter 'match-name: lo":

```
ethernets:
      ...
      match:              // this is correct - picked by the first "if" block
        name: lo
      match-name: lo //  incorrect, this line is added by the last "else" block
      ....
```

Tested locally.